source/dub/commandline.d

```diff
@@ -848,12 +849,55 @@ class BuildCommand : GenerateCommand {
 		args.getopt("f|force", &m_force, [
 			"Forces a recompilation even if the target is up to date"
 		]);
+		args.getopt("y|yes", &m_yes, [
+			`Automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively.`
+		]);
 		super.prepare(args);
 		m_generator = "build";
 	}
 
 	override int execute(Dub dub, string[] free_args, string[] app_args)
 	{
+		// single package files don't need to be downloaded, they are on the disk.
+		if (free_args.length >= 1 && !m_single) {
+			auto package_name = free_args[0];
+			auto pack = dub.packageManager.getFirstPackage(package_name);
+
+			if (!pack) {
+				bool input(string caption, bool default_value = true) {
```

Comment:
There is at least one more local definition of `input` (e.g. for `dub init`) in this file.  I'd like if we could generalize this into a reusable free function + as @WebFreak001 mentioned the `y/N` handling code.
Not a strong requirement from my side, just nice to do.

Reply:
Hmm the `dub init` `input` is quite different to this input as (1) a default value is allowed and (2) multiple values are allowed for a single value (all `yes`, `y`, `1`, `YeS`, `Y`, ...) map to `1`.

Though I'm not against unifying them by any means if someone wants to try it _after_ this addition.